Xamarin.Forms.Platform.Android/Renderers/ShellItemRendererBase.cs

```diff
@@ -115,13 +115,19 @@ protected virtual Task<bool> HandleFragmentUpdate(ShellNavigationSource navSourc
 			if (!_fragmentMap.ContainsKey(ShellSection))
 			{
 				_fragmentMap[ShellSection] = GetOrCreateFragmentForTab(ShellSection);
+				foreach (var pageItem in stack)
+				{
+					if (pageItem != null && !_fragmentMap.ContainsKey(pageItem))
+						_fragmentMap[pageItem] = CreateFragmentForPage(pageItem);
```

Comment:
Would it make sense to make this code a bit more on demand instead of looping through the stack? I'm just wondering about cases where someone inserts like 5 pages before the visible one. I don't know if we want all 5 of those creating at this point.

Reply:
I see the insert code creates the fragment
```
case ShellNavigationSource.Insert:
					if (!_fragmentMap.ContainsKey(page))
						_fragmentMap[page] = CreateFragmentForPage(page);
```

I wonder if it needs to? Just trying to be startup performance conscious :-)

Reply:
ah, I only saw the top code snippet of this comment before I pushed my latest. Let me know if you still would rather it be this way and I can adjust

Reply:
I'd say we go for the more ondemand approach if it's possible. I feel like if we don't do that now then in like a year we'll spend two days analyzing performance to realize this is an area where we can improve on.  Can you also add a stack operation that removes a page from the stack and make sure everything still works? Something like

```C#
PushAsync(page1)
PushAsync(page2)
PushAsync(page3)
Remove(page2)
GoBack
GoBack
```

Reply:
I did add a more "on demand" approach, just not in the same exact spot your comment indicated.
I'll adjust the test to include removing the inserted page per your feedback.

Reply:
Perfect! So that should let us remove this code? https://github.com/xamarin/Xamarin.Forms/pull/6740/files#diff-bd1b1bec333e9e16c2bbd309a20abfadR119

Reply:
ah, I see what I missed and why your original suggestion was better. I removed the looping but this still does some more unnecessary work than it has to in some cases. That's two strikes on naive approaches for me, perhaps I'm too heavily in vacation mode :-). I'll try to set it right this time.

Reply:
@PureWeen should be more what you were thinking now, I'll get the tests adjusted a little later